templates/terraform/iam/iam_attributes.go.erb

```diff
@@ -1,13 +1,13 @@
-<% example = object.examples.reject(&:skip_test)
+<% example = object.examples
          .reject { |e| @api.version_obj_or_closest(version) < @api.version_obj_or_closest(e.min_version) }
          .first -%>
 <% parent_resource_last_param_name = extract_identifiers(object.self_link_url).last -%>
-<% parent_resource_type_type = object.iam_policy.parent_resource_type || terraform_name -%>
+<% parent_resource_type_type = object.iam_policy.parent_resource_type || terraform_name.delete_suffix("_iam") -%>
 <% params.each_with_index do |p, i| -%>
 <% if i == params.size - 1 -%>
 <% attribute = object.iam_policy.parent_resource_attribute || parent_resource_last_param_name.underscore -%>
 <% else -%>
 <% attribute = p.underscore -%>
 <% end -%>
-	<%= p.underscore -%> = "${<%= parent_resource_type_type -%>.<%= example.primary_resource_id -%>.<%= attribute -%>}"
+	<%= attribute.underscore -%> = "${<%= parent_resource_type_type -%>.<%= example.primary_resource_id -%>.<%= attribute -%>}"
```

Comment:
I think we want to keep this as `p.underscore`. It allows us to have a `name` field on a parent resource that we change the name of for the IAM resource.

For example `backend_service_name` rather than `name` in the following:
```
resource "google_iap_web_backend_service_iam_binding" "foo" {
	project = "${google_compute_backend_service.default.project}"
	backend_service_name = "${google_compute_backend_service.default.name}"
	role = "%{role}"
	members = ["user:admin@hashicorptest.com"]
}
```

Reply:
Wouldn't that cause it to be
```
   name = "${google_compute_backend_service.default.backend_service_name}"
```
then? I actually think this should be switched p and attribute, since params is whatever is in parent object's self_link_url, and we only override it in attribute. If i leave it as is, I get:
```
name = "${google_cloudfunctions_function.function.cloud_function}"
```

instead of

```
cloud_function = "${google_cloudfunctions_function.function.name}"
```

Reply:
Ah yeah, so I did this the other way when I was doing IAP. 

For IAP I defined the URL as `base_url: 'projects/{{project}}/iap_web/compute/services/{{backendServiceName}}'`
and then use this to change the attribute to `default.name`.

That probably doesn't make sense when we can't define the URL, so you're right we should probably reverse it